panel-app/src/components/store.ts

```diff
@@ -130,22 +106,34 @@ function updateState(newState): boolean {
 				metadata &&
 				Object.keys(metadata).reduce((acc, key) => {
```

Comment:
I see this logic with `reduce` a few times in this file. Can we have a function `getFormattedMetadata()`?
They are all the same, no?
1/4